metabase/metabase
# [Blocker] Fix optional parameter handling in action forms

Fixes https://github.com/metabase/metabase/issues/28982
Demo https://www.loom.com/share/25ac0e50fddf44e685de796cf3b09e9d
Depends on `action-blank-param` branch

Changes:
- **Allows optional values to be cleared and submitted in basic action forms**
- Allow to set `Default value` only for required fields, like it's done in the QB
- Make the default value user-visible - previously because of wrong form initialization it didn't work
- Allow to submit the form if all missing parameters are optional

How to test:
- New -> Action
- Add a few parameters and make them all optional
- Make sure it is possible to run the action without providing any values from the model detail page

<img width="666" alt="Screenshot 2023-03-08 at 21 15 04" src="https://user-images.githubusercontent.com/8542534/223818099-18b0b837-dd9e-48a3-9b71-8db22fbb637c.png">


- - -
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/metabase/metabase/29049)
